### PR TITLE
M365D Hunting query improvements

### PR DESCRIPTION
**Pull request by rahul0216**
Change(s):
   - Updated name and description
   - Added version property
   - Added entity mappings

   Reason for Change(s):
   - Improving old hunting queries

   Version Updated:
   - Yes, individually for each query

   Testing Completed:
   - Yes, KQL functional testing

### CODE REVIEW

File: Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
-  | where column_ifexists('SPF','') =~ "fail" or  column_ifexists('DMARC','') =~ "fail" or column_ifexists('DKIM','') =~ "fail" or column_ifexists('CompAuth','') =~ "fail"
\ No newline at end of file
+  | where column_ifexists('SPF','') =~ "fail" or  column_ifexists('DMARC','') =~ "fail" or column_ifexists('DKIM','') =~ "fail" or column_ifexists('CompAuth','') =~ "fail"
+  | extend Account_0_Name = SenderFromAddress
+  | extend IP_0_Address = SenderIPv4
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: SenderFromAddress
```
Comment (vakohl) [inline Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml:27]: use of Name and UPNSuffix here as well

File: Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
+  | join kind=inner EmailEvents on SenderIPv4  
+  | where DeliveryAction =~ "Delivered"
+  | extend Account_0_Name = SenderFromAddress
+  | extend IP_0_Address = SenderIPv4
+  | extend MailBox_0_MailboxPrimaryAddress = RecipientEmailAddress
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: SenderFromAddress
```
Comment (vakohl) [inline Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml:29]: use of Name and UPNSuffix.

File: Solutions/Microsoft 365 Defender/Hunting Queries/Appspot Phishing Abuse.yaml [hunk 3]
```diff
@@ -23,4 +25,26 @@ query: |
...
+  | project-away NetworkMessageId1, ReportId1, Type1, TimeGenerated1, Timestamp1
+  | extend Account_0_Name = SenderFromAddress
+  | extend IP_0_Address = SenderIPv4
+  | extend URL_0_Url = Url
+  | extend MailBox_0_MailboxPrimaryAddress = RecipientEmailAddress
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: SenderFromAddress
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft 365 Defender/Hunting Queries/Appspot Phishing Abuse.yaml]: Done

File: Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml (hunk 1 above)
Comment (rahul0216) [reply Solutions/Microsoft 365 Defender/Hunting Queries/Check for spoofing attempts on the domain with Authentication failures.yaml]: Done

File: Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml (hunk 2 above)
Comment (rahul0216) [reply Solutions/Microsoft 365 Defender/Hunting Queries/Delivered Bad Emails from Top bad IPv4 addresses .yaml]: Done

File: Solutions/Microsoft 365 Defender/Hunting Queries/EmailDelivered-ToInbox.yaml [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
   - EmailEvents
 tactics:
 - InitialAccess
+relevantTechniques:
+- T1566
 query: |
   EmailEvents
-  | where isnotempty( ThreatTypes) and DeliveryLocation in ("Inbox/folder","Junk folder")
\ No newline at end of file
+  | where isnotempty(ThreatTypes) and DeliveryLocation in ("Inbox/folder","Junk folder")
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft 365 Defender/Hunting Queries/EmailDelivered-ToInbox.yaml]: Yes, good catch. Updated it.

File: Solutions/Microsoft 365 Defender/Hunting Queries/EmailDelivered-ToInbox.yaml [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
+  | extend Account_0_Name = SenderFromAddress
+  | extend IP_0_Address = SenderIPv4
+  | extend MailBox_0_MailboxPrimaryAddress = RecipientEmailAddress
+entityMappings:
+  - entityType: Account
+    fieldMappings:
+      - identifier: Name
+        columnName: SenderFromAddress
```
[the remark this answers is not in the document]
Comment (rahul0216) [reply Solutions/Microsoft 365 Defender/Hunting Queries/EmailDelivered-ToInbox.yaml]: Yes done.

**Review by vakohl — APPROVED** [review]

**Pull request merged**